### PR TITLE
[LETS-165] Finalize "unknown" server type

### PR DESCRIPTION
**Pull request by adrianbrab**
http://jira.cubrid.org/browse/LETS-165

If server boot has an error before server type is initialized, the global server type may remain unknown. When server type is finalized, this case does not require any disconnect function call.

### CODE REVIEW

File: src/base/server_type.cpp [hunk 1]
```diff
@@ -94,7 +94,7 @@ void finalize_server_type ()
     {
       ats_Gl.disconnect_page_server ();
     }
-  else
+  if (get_server_type() == SERVER_TYPE_PAGE)
```
Comment (ribram) [inline src/base/server_type.cpp]:
```suggestion
  else if (get_server_type() == SERVER_TYPE_PAGE)
```

**Review by ribram — APPROVED** [review]

File: src/base/server_type.cpp [hunk 2]
```diff
@@ -94,7 +94,7 @@ void finalize_server_type ()
     {
       ats_Gl.disconnect_page_server ();
     }
-  else
+  else if (get_server_type() == SERVER_TYPE_PAGE)
     {
       assert (get_server_type () == SERVER_TYPE_PAGE);
```
Comment (hornetmj) [inline src/base/server_type.cpp]:
```suggestion
```

Comment (adrianbrab) [reply src/base/server_type.cpp]: removed now

File: src/base/server_type.cpp [hunk 3]
```diff
@@ -99,9 +99,8 @@ void finalize_server_type ()
     {
       ats_Gl.disconnect_page_server ();
     }
-  else
+  else if (get_server_type() == SERVER_TYPE_PAGE)
     {
-      assert (get_server_type () == SERVER_TYPE_PAGE);
       ps_Gl.disconnect_active_tran_server ();
     }
```
Comment (ribram) [inline src/base/server_type.cpp:105]: Nitpicking:
```suggestion
    }
  else
    {
      assert (get_server_type () == SERVER_TYPE_UNKNOWN);
    }
```

Comment (adrianbrab) [reply src/base/server_type.cpp:105]: added now

**Pull request merged**